mod_ci/controllers.py

```diff
@@ -622,19 +621,33 @@ def start_ci():
 
 
 def update_build_badge(status, test):
+    '''
+    Build status badge for current test to be displayed on sample-platform.ccextractor.org
+
+    :param status: current testing status
+    :type status: str
+    :param test: current commit that is tested
+    :type test: str
+    :return: Nothing.
+    :rtype: None
+    '''
     if test.test_type == TestType.commit:
         parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-        output = os.path.join(parent_dir, 'static', 'svg', 'status-{platform}.svg'.format(platform=test.platform.value))
-        dwg = svgwrite.Drawing(output, profile='full')
-
+        availableon = os.path.join(parent_dir, 'static', 'svg',
+                                   '{status}-{platform}.svg'.format(platform=test.platform.value))
         if status == Status.SUCCESS:
-            dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
+            availableon.format(status, platform=test.platform.value)
+
         elif status == Status.FAILURE:
-            dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
-        else:
-            dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
-        dwg.save()
+            availableon.format(status, platform=test.platform.value)
 
+        else:
+            availableon.format(status, platform=test.platform.value)
```

Comment:
The same still applies for this location.

Besides this, a .format requires all parameters to be filled in at once, so your first format for availableon will throw a KeyError.

You can do this without any if/else structures within the outer if.